qemu/tests/tpm_verify_device.py

```diff
@@ -61,12 +71,26 @@ def search_keywords(patterns, string, flags=re.M, split_string=';'):
                     cmd = params.get('cmd_%s' % name)
                     search_keywords(pattern, session.cmd(cmd))
 
+            cmd_get_tpmsupportedfeature = params.get('cmd_get_tpmsupportedfeature')
+            output_get_tpmsupportedfeature = params.get('output_get_tpmsupportedfeature')
+            if cmd_get_tpmsupportedfeature:
+                tpmspportedfeature = session.cmd(cmd_get_tpmsupportedfeature)
+                test.log.info(tpmspportedfeature)
+                search_keywords(output_get_tpmsupportedfeature, tpmspportedfeature)
+
             reboot_method = params.get("reboot_method")
             if reboot_method:
                 error_context.context("Reboot guest '%s'." % vm.name, test.log.info)
                 vm.reboot(session, reboot_method).close()
                 continue
 
+            error_context.context("%s: Check Windows event log inside guest after test." % vm.name,
+                                  test.log.info)
+            if cmd_check_event_log:
+                event_log = session.cmd_output(cmd_check_event_log)
+                test.log.info(event_log)
+                search_keywords(event_log_pattern, event_log)
```

Comment:
This is redundant for me. You have 3 similar implementation, we need a function for it then, e.g.
```
def check_error_msg(chk_cmd, chk_pattern):
    if not chk_cmd or not chk_pattern:
        return
    full_msg = session.cmd_output(chk_cmd)
    test.log.info(full_msg)
    search_keywords(chk_pattern, full_msg)
```
```
check_error_msg(cmd_check_event_log, event_log_pattern)
```
```
check_error_msg(cmd_get_tpmsupportedfeature, tpmspportedfeature)
```

Reply:
If I understand this comment correctly, search_keywords() is the function. And it is called at L64, L79 and L92 for different patterns. Or maybe I did not understand the point? 

I removed some duplicated printing log infos now.